app/services/order_workflow.rb

```diff
@@ -21,6 +21,14 @@ def next(options = {})
     result
   end
 
+  def advance_to_payment
+    until order.state == "payment"
+      break unless order.next
+
+      after_transition_hook(advance_order_options)
+    end
+  end
+
```

Comment:
this is copy pasted from advance_order, right? can youcreate a method advance_to_state(target_state) and reuse here and in advance_order?

Reply:
👍 f27c58ac7b240bd4d204257568a619827a9c1ea0